src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -1302,12 +1304,17 @@ private IHtmlContent GenerateGroupsAndOptions(string optionLabel, IEnumerable<Se
             foreach (var group in groupedSelectList)
             {
                 var optGroup = group.First().Group;
-
-                // Wrap if requested.
-                TagBuilder groupBuilder = null;
                 if (optGroup != null)
                 {
-                    groupBuilder = new TagBuilder("optgroup");
+                    var optGroupContent = new BufferedHtmlContent();
```

Comment:
This is used in <a href="https://github.com/aspnet/Mvc/pull/2952/files#diff-0e83439eeed8ffbde4eceb0db1f6e541R1314">line 1314</a>.